src/react/components/pages/editorPage/editorPage.test.tsx

```diff
@@ -191,13 +194,105 @@ describe("Editor Page Component", () => {
             expect.objectContaining({
                 asset: {
                     ...expectedAsset,
-                    state: AssetState.Tagged,
+                    state: AssetState.Visited,
                 },
             }),
         );
         expect(saveProjectSpy).toBeCalledWith(expect.objectContaining(partialProject));
     });
 
+    it("sets page state to invalid when edited asset includes un-tagged regions", async () => {
+        // create test project and asset
+        const testProject = MockFactory.createTestProject("TestProject");
+        const defaultAsset = testAssets[0];
+
+        // mock store and props
+        const store = createStore(testProject, true);
+        const props = MockFactory.editorPageProps(testProject.id);
+
+        const saveAssetMetadataSpy = jest.spyOn(props.actions, "saveAssetMetadata");
+        const saveProjectSpy = jest.spyOn(props.actions, "saveProject");
+
+        // create mock editor page
+        const wrapper = createComponent(store, props);
+        const editorPage = wrapper.find(EditorPage).childAt(0) as ReactWrapper<IEditorPageProps, IEditorPageState>;
+
+        await MockFactory.flushUi();
+        wrapper.update();
+
+        // Create a new un-tagged region
+        const newRegion = MockFactory.createTestRegion("unTaggedRegion", []);
+        const assetMetadata: IAssetMetadata = {
+            asset: defaultAsset,
+            regions: [newRegion],
+            version: appInfo.version,
+        };
+
+        saveAssetMetadataSpy.mockClear();
+        saveProjectSpy.mockClear();
+
+        // Initial state change of region
+        wrapper.find(Canvas).props().onAssetMetadataChanged(assetMetadata);
+
+        expect(editorPage.state().isValid).toBe(false);
+        expect(saveAssetMetadataSpy).not.toBeCalled();
+        expect(saveProjectSpy).not.toBeCalled();
+
+        // Apply tag to region
+        newRegion.tags = ["test"];
+        wrapper.find(Canvas).props().onAssetMetadataChanged(assetMetadata);
+
+        await MockFactory.flushUi();
+
+        expect(editorPage.state().isValid).toBe(true);
```

Comment:
the test description is about testing invalid state but the test seems to be testing valid state

Reply:
If you look through the whole test it tests both conditions.  The page goes into an invalid state, gets fixed and verifies it ends up in a valid state